### PR TITLE
Dilpreet | Test | Fixed exceptions

### PR DESCRIPTION
**Pull request by ravinderwalkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

**Review by the-prateek-mishra — APPROVED** [review]

File: apps/web-giddh/src/app/material-ledger/ledger.component.ts [hunk 1]
```diff
@@ -377,8 +377,8 @@ export class LedgerComponent implements OnInit, OnDestroy {
             This logic is only required in ledger.
         */
         const accountUniqueName = e.additional.stock && (currentLedgerCategory === 'income' || currentLedgerCategory === 'expenses') ?
-            this.lc.activeAccount ? this.lc.activeAccount.uniqueName : '' :
-            e.additional?.uniqueName;
+            this.lc.activeAccount ? this.lc.activeAccount?.uniqueName : '' :
+            e.additional.uniqueName;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/material-ledger/ledger.component.ts]:
```suggestion
            e.additional?.uniqueName;
```

File: apps/web-giddh/src/app/material-ledger/ledger.component.ts [hunk 2]
```diff
@@ -377,8 +377,8 @@ export class LedgerComponent implements OnInit, OnDestroy {
             This logic is only required in ledger.
         */
         const accountUniqueName = e.additional.stock && (currentLedgerCategory === 'income' || currentLedgerCategory === 'expenses') ?
-            this.lc.activeAccount ? this.lc.activeAccount.uniqueName : '' :
-            e.additional?.uniqueName;
+            this.lc.activeAccount ? this.lc.activeAccount?.uniqueName : '' :
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/material-ledger/ledger.component.ts]:
```suggestion
            this.lc.activeAccount ? this.lc.activeAccount.uniqueName : '' :
```

File: apps/web-giddh/src/app/material-ledger/ledger.component.ts [hunk 3]
```diff
@@ -546,7 +546,7 @@ export class LedgerComponent implements OnInit, OnDestroy {
                             currentBranchUniqueName = this.generalService.currentBranchUniqueName;
                             this.currentBranch = _.cloneDeep(response.find(branch => branch?.uniqueName === currentBranchUniqueName)) || this.currentBranch;
                         } else {
-                            currentBranchUniqueName = this.activeCompany ? this.activeCompany.uniqueName : '';
+                            currentBranchUniqueName = this.activeCompany ? this.activeCompany?.uniqueName : '';
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/material-ledger/ledger.component.ts]:
```suggestion
                            currentBranchUniqueName = this.activeCompany ? this.activeCompany.uniqueName : '';
```

File: apps/web-giddh/src/app/material-ledger/ledger.component.ts [hunk 4]
```diff
@@ -1467,7 +1467,7 @@ export class LedgerComponent implements OnInit, OnDestroy {
             const currentLedgerCategory = this.lc.activeAccount ? this.generalService.getAccountCategory(this.lc.activeAccount, this.lc.activeAccount.uniqueName) : '';
             // If current ledger is of income or expense category then send current ledger as stockAccountUniqueName. Only required for ledger.
             const accountUniqueName = (currentLedgerCategory === 'income' || currentLedgerCategory === 'expenses') ?
-                this.lc.activeAccount ? this.lc.activeAccount.uniqueName : '' :
+                this.lc.activeAccount ? this.lc.activeAccount?.uniqueName : '' :
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/material-ledger/ledger.component.ts:1465]:
```suggestion
                this.lc.activeAccount ? this.lc.activeAccount.uniqueName : '' :
```

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 5]
```diff
@@ -1475,8 +1475,8 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
                         }
                         if (this.currentVoucherFormDetails?.depositAllowed || (this.isPendingVoucherType && obj.accountDetails && obj.voucherDetails)) {
                             if (!this.isLastInvoiceCopied) {
-                                obj.accountDetails.name = (this.voucherApiVersion === 2 && this.isCashInvoice && results[0].account.customerName) ? results[0].account.customerName : results[0].account.name;
-                                obj.voucherDetails.customerName = (this.voucherApiVersion === 2 && this.isCashInvoice && results[0].account.customerName) ? results[0].account.customerName : results[0].account.name;
+                                obj.accountDetails.name = (this.voucherApiVersion === 2 && this.isCashInvoice && results[0].account.customerName) ? results[0].account.customerName : results[0].account?.name;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts]:
```suggestion
                                obj.accountDetails.name = (this.voucherApiVersion === 2 && this.isCashInvoice && results[0].account?.customerName) ? results[0].account?.customerName : results[0].account?.name;
```

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 6]
```diff
@@ -1475,8 +1475,8 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
...
+                                obj.voucherDetails.customerName = (this.voucherApiVersion === 2 && this.isCashInvoice && results[0].account.customerName) ? results[0].account.customerName : results[0].account?.name;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts]:
```suggestion
                                obj.voucherDetails.customerName = (this.voucherApiVersion === 2 && this.isCashInvoice && results[0].account?.customerName) ? results[0].account?.customerName : results[0].account?.name;
```

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 7]
```diff
@@ -5076,7 +5076,7 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
                     if (taxTypeArr.indexOf(ta.taxType) > -1) {
                         salesEntryClass.isOtherTaxApplicable = true;
                         let otherTaxModal = new SalesOtherTaxesModal();
-                        otherTaxModal.appliedOtherTax = { name: ta.name, uniqueName: ta.uniqueName };
+                        otherTaxModal.appliedOtherTax = { name: ta?.name, uniqueName: ta.uniqueName };
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts]:
```suggestion
                        otherTaxModal.appliedOtherTax = { name: ta?.name, uniqueName: ta?.uniqueName };
```

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 8]
```diff
@@ -5094,7 +5094,7 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
                             isChecked: true,
                             isDisabled: false,
                             type: ta.taxType,
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts]:
```suggestion
                            type: ta?.taxType,
```

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 9]
```diff
@@ -6261,7 +6261,7 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
                         const results = response.map(result => {
                             return {
                                 value: result.stock ? `${result.uniqueName}#${result.stock.uniqueName}` : result.uniqueName,
-                                label: result.stock ? `${result.name} (${result.stock.name})` : result.name,
+                                label: result.stock ? `${result?.name} (${result.stock?.name})` : result?.name,
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts]:
```suggestion
                                label: result.stock ? `${result.name} (${result.stock?.name})` : result.name,
```

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 10]
```diff
@@ -6366,7 +6366,7 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
         const searchResults = results.map(result => {
             return {
                 value: result.stock ? `${result.uniqueName}#${result.stock.uniqueName}` : result.uniqueName,
-                label: result.stock ? `${result.name} (${result.stock.name})` : result.name,
+                label: result.stock ? `${result?.name} (${result.stock?.name})` : result?.name,
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts]:
```suggestion
                label: result.stock ? `${result.name} (${result.stock?.name})` : result.name,
```

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 11]
```diff
@@ -7146,7 +7146,7 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
             const results = response.map(result => {
                 return {
                     value: result.stock ? `${result.uniqueName}#${result.stock.uniqueName}` : result.uniqueName,
-                    label: result.stock ? `${result.name} (${result.stock.name})` : result.name,
+                    label: result.stock ? `${result?.name} (${result?.stock?.name})` : result?.name,
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts]:
```suggestion
                    label: result.stock ? `${result.name} (${result.stock?.name})` : result.name,
```

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 12]
```diff
@@ -7163,7 +7163,7 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
             this.defaultItemSuggestions = response.map(result => {
                 return {
                     value: result.stock ? `${result.uniqueName}#${result.stock.uniqueName}` : result.uniqueName,
-                    label: result.stock ? `${result.name} (${result.stock.name})` : result.name,
+                    label: result.stock ? `${result?.name} (${result.stock?.name})` : result?.name,
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts]:
```suggestion
                    label: result.stock ? `${result.name} (${result.stock.name})` : result.name,
```

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 13]
```diff
@@ -7320,8 +7320,8 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
             }
             data.state.code = (isCompanyAddress) ? address.stateCode : (address.state) ? address.state.code : "";
             data.stateCode = data.state.code;
-            data.state.name = (isCompanyAddress) ? address.stateName : (address.state) ? address.state.name : "";
-            data.stateName = data.state.name;
+            data.state.name = (isCompanyAddress) ? address.stateName : (address.state) ? address.state?.name : "";
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts]:
```suggestion
            data.state.name = (isCompanyAddress) ? address.stateName : (address.state) ? address.state.name : "";
```

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 14]
```diff
@@ -7320,8 +7320,8 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
...
+            data.stateName = data.state?.name;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts]:
```suggestion
            data.stateName = data.state.name;
```

File: apps/web-giddh/src/app/store/authentication/authentication.reducer.ts [hunk 15]
```diff
@@ -382,7 +382,7 @@ export function AuthenticationReducer(state: AuthenticationState = initialState,
                         isLoginWithPasswordIsShowVerifyOtp: true,
                         user: res.body
                     });
-                } else if (!res.body.user.isVerified) {
+                } else if (!res.body?.user.isVerified) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/store/authentication/authentication.reducer.ts]:
```suggestion
                } else if (!res.body?.user?.isVerified) {
```

**Review by ravinderwalkover — COMMENTED** [review]
Solve the mentioned issue

**Pull request closed without merging**